Experimental basic support for Teknopoint A/C protocol

crankyoldgit opened the pull request:
* Basic support added via `sendTeknopoint()` & `decodeTeknopoint()`
* Unit test coverage for the additions/changes.
* Minor code style clean-up.
* Checksum validation.

Note: Bit ordering now determined (_LSBF_) via Temp range & checksum calc.

For #1486

crankyoldgit committed:
Experimental support for Teknopoint A/C protocol
* Basic support added via `sendTeknopoint()` & `decodeTeknopoint()`
* Unit test coverage for the additions/changes.
* Minor code style cleanup.

Note: Bit ordering not yet determined. May change without notice.

For #1486

crankyoldgit committed:
Update supported remote model info

crankyoldgit committed:
Tweak timings & matching tolerance.
For https://github.com/crankyoldgit/IRremoteESP8266/issues/1486#issuecomment-852446565

crankyoldgit committed:
Reverse bit order & add a checksum test
* Convert to LSBF order.
* Update unit tests accordingly.
* Add a checksum test for the protocol.

For #1486

darshkpatel reviewed (approved):
LGTM 👍🏻

NiKiZe commented on src/IRremoteESP8266.h:
```diff
@@ -766,7 +773,7 @@
      DECODE_MITSUBISHI112 || DECODE_HITACHI_AC424 || DECODE_HITACHI_AC3 || \
      DECODE_HITACHI_AC344 || DECODE_CORONA_AC || DECODE_SANYO_AC || \
-     DECODE_VOLTAS || DECODE_MIRAGE || DECODE_HAIER_AC176)
+     DECODE_VOLTAS || DECODE_MIRAGE || DECODE_HAIER_AC176 || DECODE_TEKNOPOINT)
```
This line looks a bit on the long end, even if it is ok now, should we have the foresight to add it on a new line?
Should we also end with a || \\nfalse to never have to care about that part ever changing?

crankyoldgit replied:
Oooo. I like the `|| false` idea. That's a great one. Will update it.

crankyoldgit replied:
Done.

crankyoldgit committed:
Improve layout of the `DECODE_AC` define.

Per excellent code review feedback in
https://github.com/crankyoldgit/IRremoteESP8266/pull/1504#discussion_r658188100
